test/compress/dead-code.js

```diff
@@ -68,10 +68,9 @@ dead_code_constant_boolean_should_warn_more: {
         side_effects : true,
     };
     input: {
-        while (!((foo && bar) || (x + "0"))) {
+        while (!(foo || (x + "0"))) {
             console.log("unreachable");
             var foo;
-            function bar() {}
```

Comment:
> parse() doesn't have ecma mode switches AFAICT.

I appreciate that, but is there a need for `ecma` in parse and compress? Is there ambiguity that could only be resolved via such an explicit option?